enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,513 @@
+
+In the future, at some degree, [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+approach might help to solve the issue and give a room for supplementing *some* platform-dependent components, however some integral parts of
+OpenShift can not be disabled and still requires a signal about an underlying platform, for example, KCM and MCO, in respect of the enablement of
+[external cloud controller manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/).
+
+One of the possible examples of the interaction of the capabilities with the "External" platform type would be the MachineAPI and the Machine Api Operator.
+At this point, the MachineAPI has no use in running if there is no machine controller (which is heavily cloud-provider dependent).
+When the platform type is set to "External", and the machine-api capability is enabled,
+that will cause Machine API operator to deploy only generic cloud-independent controllers (such as machine-healthcheck, machineset, and node-link controllers).
+The platform-specific components would be deployed through a separate mechanism.
+Such behaviour will simplify initial cloud-platform enablement and will reduce the necessity of reverse-engineering
+and replicating work that was already done by Red Hat engineers.
+
+### User Stories
+
+- As a cloud-provider affiliated engineer / platform integrator / RH partner
+  I want to have a mechanism to signal Openshift's built-in operators about additional
+  cloud-provider specific components so that I can inject my own platform-specific controllers into OpenShift
+  to improve the integration between OpenShift and my cloud provider.
+
+  We are aware of examples of supplementing OKD installations with custom machine-api controllers,
+  however, users are experiencing a lot of difficulties on this path due to the necessity of, literally, reverse engineering,
+  manual management of generic MAPI controllers, and so on.
+
+- As a cloud provider whose platform is not integrated into OpenShift,
+  I want to have the Cloud Controller Manager for my infrastructure running in OpenShift from the initial install.
+  Having a platform type that allows for the addition of operators or components which perform platform-specific functionality would help
+  me to create better integrations between OpenShift and my infrastructure platform.
+
+### Goals
+
+- Reduce the necessity to make changes in "openshift/api", "openshift/library-go" and dependant infrastructure-related components during basic cloud-provider integration with OpenShift
+- Make a cloud provider integration process more accessible and simple to external contributors as well as for RH engineers
+- Provide a top-level description of the operation mode for affected components at the first steps
+- Introduce a somewhat "neutral" platform type, which would serve as a signal about an underlying cloud-provider presence
```

Comment:
Perhaps an `underlying but generic cloud-provider presence` to help signal that this is a generalised approach for the long tail of clouds?

Reply:
should there be some commas?

eg
> ... underlying, but generic, cloud-provider presence

Reply:
Yes I would say there should be